src/Service/RequestsQueue.h

```diff
@@ -51,14 +51,16 @@ struct RequestsQueue
         }
     }
 
-    bool push(const KeeperStore::RequestForSession & request)
+    template <typename Request>
+    bool push(Request && request)
```

Comment:
We should keep ` bool push(const KeeperStore::RequestForSession & request)`

Reply:
Use Universal Reference is better.

Reply:
We'd better remove template form `push` to keep code consistancy. And we can just add `push(&&) `.

Reply:
No, the code is cleaner this way.

Reply:
Because elsewhere T is the class template parameter. In this method the parameter is not a template type, so you can write it like this.

Reply:
Just code style issue, not importent.